### PR TITLE
Update `hyper-rustls` and fix lints

### PR DESCRIPTION
**Pull request by KaDiWa4**
hyper-rustls now has a builder type and the old `with_native_certs` is gone.
HTTP 2 (h2) is now disabled by default.

### CODE REVIEW

**Comment by CryZe** [comment]
Does that mean we are not using HTTP2 anymore when talking to splits.io? I think I'd prefer HTTP 2 over HTTP 1.

**Comment by KaDiWa4** [comment]
I'll add it back in then

File: Cargo.toml [hunk 1]
```diff
@@ -28,10 +28,10 @@ url = "2.1.0"
 uuid = { version = "0.8.1", default-features = false, features = ["serde"] }
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
-hyper = { version = "0.14.1", default-features = false, features = ["tcp", "client", "http1", "http2"] }
+hyper = { version = "0.14.1", default-features = false, features = ["tcp", "client", "http2"] }
 
 [target.'cfg(all(any(target_os = "linux", target_family = "windows", target_os = "macos"), any(target_arch = "x86", target_arch = "x86_64", target_arch = "arm", target_arch = "aarch64")))'.dependencies]
-hyper-rustls = "0.23.0"
+hyper-rustls = { version = "0.23.0", default-features = false, features = ["native-tokio", "http2", "tls12", "logging"] }
```
Comment (CryZe) [inline Cargo.toml:34]: Why is logging required?

Comment (KaDiWa4) [reply Cargo.toml:34]: it isn't, it's one of the defaults so I thought I'd include it

Comment (CryZe) [reply Cargo.toml]: In general we'd want to disable as many defaults as possible in order to not force any upon our dependents.

**Comment by CryZe** [comment]
Looks good now, thanks for working on this. 👍

**Pull request merged**